lib/parameter_parser/search_parameter_parser.rb

```diff
@@ -101,6 +103,37 @@ def capped_count
     end
   end
 
+  def parse_query(query)
+    quoted = []
+    unquoted = []
+
+    current_keyword = ''
+    is_inside_quote = false
+    query.split('').each do |c|
+      if c == '"'
+        current_keyword = current_keyword.strip
+        if is_inside_quote
+          quoted << current_keyword unless current_keyword.empty?
+        else
+          unquoted << current_keyword unless current_keyword.empty?
+        end
+        is_inside_quote = !is_inside_quote
+        current_keyword = ''
+      else
+        current_keyword << c
+      end
+    end
```

Comment:
Could use a regex here, but I spent too long trying to get an array of unquoted strings without success. Your solution is better!

```
quoted = '"S" "A" "s" z y x "'.scan(/"([^"]*)"/).flatten
```

Reply:
Well, you know what they say about regex...

> Some people, when confronted with a problem, think "I know, I'll use regular expressions." Now they have two problems.